test/App/Fossa/API/BuildLinkSpec.hs

```diff
@@ -12,21 +12,21 @@ import Data.Text (Text)
 import Diag.Result (resultToMaybe)
 import Fossa.API.Types
 import Srclib.Types (Locator (Locator))
-import Test.Hspec
-import Text.URI.QQ
+import Test.Hspec (Spec, describe, it, shouldBe)
+import Text.URI.QQ (uri)
 
 simpleSamlPath :: Text
-simpleSamlPath = "https://app.fossa.com/account/saml/1?next=/projects/fetcher123%2bproject123/refs/branch/master123/revision123"
+simpleSamlPath = "https://app.fossa.com/account/saml/1?next=/projects/fetcher123%252bproject123/refs/branch/master123/revision123"
 
 -- | Note the differences here between '%2F' and '%252F'.  The percent sign is re-encoded so that it's properly handled on the next redirect.
 gitSamlPath :: Text
-gitSamlPath = "https://app.fossa.com/account/saml/103?next=/projects/fetcher@123%252fabc%2bgit@github.com%252fuser%252frepo/refs/branch/weird--branch/revision@123%252fabc"
+gitSamlPath = "https://app.fossa.com/account/saml/103?next=/projects/fetcher%2540123%252fabc%252bgit%2540github.com%252fuser%252frepo/refs/branch/weird--branch/revision%2540123%252fabc"
```

Comment:
We should probably update the comment to be more general about the %25 double-encoding.

Reply:
I'm updated the comment describing what's going on here.